CREATING_TOXICS.md

```diff
@@ -115,6 +115,24 @@ The unit used by `GetBufferSize()` is `StreamChunk`s. Chunks are generally anywh
 1 byte, up to 32KB, so keep this in mind when thinking about how much buffering you need,
 and how much memory you are comfortable with using.
 
+## Stateful toxics
+
+If a toxic needs to store extra information for a connection such as the number of bytes
+transferred (See `limit_data` toxic), a state object can be created by implementing the
+`StatefulToxic` interface. This interface defines the `NewState()` function that can create
```

Comment:
This way the general / common case is simpler. Most toxics are stateless, so they would all have a `NewState()` function that just returns `nil`. This way the function is optional. It operates the same way as `BufferedToxic`, which defaults to a buffer size of 0.